src/Query/QueryResult.php

```diff
@@ -485,13 +485,35 @@ public function toArray() {
 			'meta'=> [
 				'hash'   => md5( json_encode( $serializeArray ) ),
 				'count'  => $this->getCount(),
+				'total'  => $this->getTotalCount(),
 				'offset' => $this->mQuery->getOffset(),
 				'source' => $this->mQuery->getQuerySource(),
 				'time'   => number_format( ( microtime( true ) - $time ), 6, '.', '' )
 				]
 			]
 		);
 	}
+	
+	/**
+	 * Returns the total number of query results regardless of maximum to retrieval
+	 * 
+	 * @since ...
+	 *
+	 * @return integer|null
+	 */
+	public function getTotalCount() {
+		$query = $this->mQuery;
+		$query->querymode = \SMWQuery::MODE_COUNT; //change query format to count
+		$store = $this->getStore();
+		try {
+			$test = $store->getQueryResult( $query );
+		} catch (\Exception|\Throwable $e) {
+			return null;
```

Comment:
I'm dubious about the return value being nullable. That makes things more difficult for the callers.

Catching exceptions also makes it impossible for them to handle exceptions in the way they want.

Is there some reason to not always return an int, and to let exceptions bubble up to the caller?

Reply:
My main concern here is/was that queries without support for the `count` format, such as queries produced by Semantic Compound Queries and possibly others I'm not aware of, should still be able to produce Api data, without `ApiQueryResultFormatter::doFormat` throwing an error message instead. 

On the other hand, you could argue that relying on an error is a little hackish and unsound if the use case (SCQ) is legitimate, which is why I hoped there might be a cleaner solution that actually checks in advance if the count format is supported for a particular query. That and you're quite right to point out that there may well be a genuine errors that would fail to get exposed if the output is nullable.  

Do you have any solution in mind?

Reply:
Returning 0 for "unsupported formats" seems like a better move than returning null.

Why would compound query results cause an error/exception?

Reply:
I suspect the reason the method wouldn’t work for SCQ is because the query object available at this point is unsuitable, e.g. it doesn’t have an equivalent query condition if you look at the serialized version. That's probably why `$this->getStore()->getQueryResult( $query );` fails.[0] 
Instead, I ended up with an approach not too dissimilar from what’s been done for the `#rowcount` parameter, creating a new query through calls to SMWQueryProcessor class methods. Still hesitant because SCQ comes with its own query processor. 

It works in my test cases, but I'm not a PHP developer and I’m not familiar enough with the architecture to know whether those calls run the risk of circularity. Please let me know if you think this is the wrong place in the code to retrieve and add a result count (I actually started by editing Query.php and ApiQueryResultFormatter.php, but then reconsidered because it seemed to me late-stage intervention).

[0]`Exception caught: Argument 1 passed to SMW\\SQLStore\\QueryEngine\\ConditionBuilder::buildFromDescription() must be an instance of SMW\\Query\\Language\\Description, null given, called in .../extensions/SemanticMediaWiki/src/SQLStore/QueryEngine/ConditionBuilder.php on line 211`